tardis/rest/app/security.py

```diff
@@ -1,127 +1,119 @@
+import secrets
 from ...configuration.configuration import Configuration
 from ...exceptions.tardisexceptions import TardisError
 
 from bcrypt import checkpw, gensalt, hashpw
-from fastapi import Depends, HTTPException, status
-from fastapi.security import OAuth2PasswordBearer, SecurityScopes
-from jose import JWTError, jwt
-from pydantic import BaseModel, ValidationError
+from fastapi import HTTPException, status, Depends
+
+from fastapi.security import SecurityScopes
+
+from pydantic import BaseModel
+from fastapi_jwt_auth import AuthJWT
 
-from datetime import datetime, timedelta
 from functools import lru_cache
-from typing import List, Optional
+from typing import Optional, List
 
 
-class TokenData(BaseModel):
-    user_name: Optional[str] = None
-    scopes: List[str] = []
+class Settings(BaseModel):
+    authjwt_secret_key: str = secrets.token_hex(128)
+    authjwt_token_location: set = {"cookies"}
+    authjwt_cookie_samesite: str = "strict"
+    # TODO: change this to true in production so only https traffic is allowed
+    authjwt_cookie_secure: bool = False
+    # TODO: Set this too to True. But as soon as possible.
+    # Update: As 'same_site' is strict this is probably enough.
+    authjwt_cookie_csrf_protect: bool = False
```

Comment:
What prevents you to set it to `True`?

Reply:
Which one do you mean? The `secure` or the `csrf` option. I didn't set `secure` because we aren't in production yet and TLS makes things more difficult during development. I set `csrf`-tokens to false because it would mean more code in the front-end and as mentioned in the comment `same-site` is enough. As I understood it, only when one deals with CDN or user supplied date, `strict` isn't enough

Reply:
The `csrf` one. Okay. I think the production setup potentially might be nginx used as proxy-> to something else. So, the TLS things will be absorbed by nginx anyhow.